build.gradle

```diff
@@ -1,33 +1,52 @@
 // See https://github.com/JetBrains/gradle-intellij-plugin/
 plugins {
-    id 'org.jetbrains.intellij' version '1.17.4'
+    id("org.jetbrains.intellij.platform") version "2.0.1"
     id 'java'
     id 'com.diffplug.spotless' version '6.25.0'
 }
 
 group 'edu.kit.kastel.sdq'
 version '1.0-SNAPSHOT'
 
+java {
+    sourceCompatibility = "21"
+    targetCompatibility = "21"
+}
+
 repositories {
-    mavenCentral()
     mavenLocal()
+    mavenCentral()
+    intellijPlatform {
+        defaultRepositories()
+    }
+    maven {
+        url 'https://s01.oss.sonatype.org/content/repositories/snapshots'
+    }
 }
 
 dependencies {
-    implementation 'edu.kit.kastel.sdq:artemis4j:6.7.5'
-    implementation 'org.eclipse.jgit:org.eclipse.jgit:6.10.0.202406032230-r'
+    intellijPlatform {
+        intellijIdeaCommunity('2024.2.1')
+        pluginVerifier()
+        bundledPlugin("com.intellij.java")
+        bundledPlugin("org.jetbrains.idea.maven")
+        instrumentationTools()
+    }
+    implementation 'edu.kit.kastel.sdq:artemis4j:7.4.2-SNAPSHOT'
     implementation 'org.apache.commons:commons-io:1.3.2'
     implementation 'de.firemage.autograder:autograder-cmd:' + AUTOGRADER_VERION
     implementation 'de.firemage.autograder:autograder-core:' + AUTOGRADER_VERION
     implementation 'de.firemage.autograder:autograder-extra:' + AUTOGRADER_VERION
```

Comment:
I think we can remove the dependencies on `-cmd` and `-core`. Only the `-extra` one is necessary, everything else is already included in artemis4j?

Reply:
I'd also move all deps to artemis4j otherwise it might be hard to sync the different versions.

Reply:
> I'd also move all deps to artemis4j otherwise it might be hard to sync the different versions.

Hm, yeah, when I think about it, we don't need any autograder dependency. I was thinking of the old implementation